### PR TITLE
Make sure the audit mode output is reflected both in the logs and in the formatter output.

### PR DESCRIPTION
**Pull request by sersut**
Fixes https://github.com/chef/chef/issues/2910

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @Sgtpluck

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

File: lib/chef/audit/logger.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+require 'stringio'
+
+class Chef
+  class Audit
+    class Logger
+      def self.puts(message="")
+        if @buffer.nil?
+          @buffer = StringIO.new
+        end
```
Comment (mcquin) [inline lib/chef/audit/logger.rb]: Ruby lets you do this: `@buffer ||= StringIO.new`, which is equivalent to this if block.

Comment (sersut) [inline lib/chef/audit/logger.rb]: Nice catch :smile: :+1:

**Comment by tyler-ball** [comment]
:+1:

**Comment by lamont-granquist** [comment]
:shipit:

**Comment by sersut** [comment]
@btm / @thommay I have rebased this and fixed a few merge conflicts. Once Travis is happy it should be ready to go.

**Comment by sersut** [comment]
Looks like the travis failure is a timeout that is not related to my changes. I have restarted the failing job. Hoping everything will turn green soon.

**Comment by sersut** [comment]
Rebased one more time. Can we get this merged in when Travis is happy?

**Pull request merged**